src/Pact/Gas/Table.hs

```diff
@@ -248,10 +254,13 @@ tableGasModel gasConfig =
           Just cs -> _gasCostConfig_selectColumnCost gasConfig * fromIntegral (length cs)
         GSortFieldLookup n ->
           gasToMilliGas $ fromIntegral n * _gasCostConfig_sortFactor gasConfig
-        GConcatenation i j ->
-          gasToMilliGas $ fromIntegral (i + j) * _gasCostConfig_concatenationFactor gasConfig
-        GFoldDB ->
-          gasToMilliGas $ _gasCostConfig_foldDBCost gasConfig
+        GConcatenation i j -> gasToMilliGas $
+          fromIntegral (i + j) * _gasCostConfig_concatenationFactor gasConfig
+        GTextConcatenation nChars nStrings -> gasToMilliGas $
+          _gasCostConfig_textConcatenationFactorOffset gasConfig +
+          (fromIntegral nChars  * _gasCostConfig_textConcatenationFactorStringLen gasConfig) `div` 1000 `div` fromIntegral nStrings +
```

Comment:
Where is the 1000 coming from?

Reply:
Also adding parens would be lovely for my sanity

Reply:
1000's removed by porting to MilliGas, I'm happy to report!